doc/md/canister-maintenance/upgrades.md

```diff
@@ -0,0 +1,154 @@
+---
+sidebar_position: 3
+---
+
+# Stable variables and upgrade methods
+
+## Overview
+
+One key feature of ICP is its ability to persist canister state using WebAssembly memory and globals rather than a traditional database. This means that the entire state of a canister is restored before and saved after each message, without explicit user instruction. This automatic and user-transparent preservation of state is called orthogonal persistence.
+
+Orthogonal persistence allows a canister to save and restore data to dedicated **stable memory** that, unlike ordinary canister memory, is retained across an upgrade and allows a canister to transfer data in bulk across upgrades.
+
+Motoko provides high-level support for preserving state that leverages stable memory. This feature, called **stable storage**, is designed to accommodate changes to both the application data and the Motoko compiler used to produce the application code.
+
+Utilizing stable storage depends the developer anticipating and indicating the data to retain after an upgrade. Depending on the application, the data you decide to persist might be some, all, or none of a given actor’s state.
```

Comment:
I think the shortened overview is actually less clear than the original and conflates orthogonal persistence with stable memory, which is misleading.
In CS orthogonal persistence refers to  completely transparent preservation of state without any user intervention/special coding. This is what the IC provides to canister between messages - the state of the canister is transparently restored before the message and saved after without the user having to read/write stuff to a database as in traditional 3-tier (frontend/backend/DB) system. Instead, all data is just stored in memory. That's the OP bit.

Upgrades actually break the illusion of OP because the user has to explicitly arrange for the state of the old canister to be saved from the old version and then restored in the new version of the canister.  In Motoko, they can do this with reasonable convenience by explicitly declaring stable variables, or less conveniently by explicitly using stable memory or regions. But as soon as explicit measures are required, it's not really orthogonal persistence.

Reply:
adressed in https://github.com/dfinity/motoko/pull/4512